ipld/blockstore/src/buffered.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::BlockStore;
+use cid::{multihash::MultihashDigest, Cid};
+use db::{Error, Store};
+use encoding::{ser::Serialize, to_vec};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::error::Error as StdError;
+
+/// Wrapper around `BlockStore` to limit and have control over when values are written.
+/// This type is not threadsafe and can only be used in syncronous contexts.
+#[derive(Debug)]
+pub struct BufferedBlockStore<'bs, BS> {
+    base: &'bs BS,
+    write: RefCell<HashMap<Cid, Vec<u8>>>,
+}
+
+impl<'bs, BS> BufferedBlockStore<'bs, BS>
+where
+    BS: BlockStore + Store,
```

Comment:
Curious what are the consequences of removing the requirement of Store on BS, since BlockStore inherets the Store trait... I removed it and it works.

Reply:
functionally the same, but yeah should be more succinct, I'll fix

Reply:
Cool. Just wanted to make sure i was on the same page. But yeah, removing is best